### ISSUE TITLE
False positive type checking error on unpacked tuple type variable

### ISSUE THREAD

**Issue by austinyu** [astral-sh/ty#4044]
### Summary

The following example throws type error in ty@0.0.61. This should not be a type checking error because mypy allow this and this is compatible with the usage of unpacked type variables. 

```
Argument is incorrect: Expected `(*args: Unknown) -> None`, found `def on_new_path(path: str) -> None` (invalid-argument-type) [Ln 8, Col 1]
```

```python
from collections.abc import Callable


def subscribe[*Ts](*states: *Ts) -> Callable[[Callable[[*Ts], None]], None]:
    raise NotImplementedError


@subscribe("some_string")
def on_new_path(path: str) -> None:
    pass

@subscribe("some_string", 123)
def on_new(path: str, some_int: int) -> None:
    pass

```
[Playground](https://play.ty.dev/8940d899-c685-47e6-ab86-05cbab175570)

### Version

0.0.61

**Comment by sharkdp** [comment]
FYI @dhruvmanila

**Comment by sharkdp** [comment]
@carljm Thanks, I hadn't seen #4011 and it seems very much related. But it seems that something else is wrong here?
 
The error message (*"Expected `(*args: Unknown) -> None`, found `def on_new_path(path: str) -> None`"*) itself seems problematic to me? Why would we not allow that function literal to be assignable to `(*args: Unknown) -> None`?

I don't know anything about the internal representation of TypeVarTuple, so maybe that is also related to some missing TypeVarTuple feature, but it looks strange?

**Comment by carljm** [comment]
You're right, that does look odd. I based my identification as duplicate mostly on the fact that https://github.com/astral-sh/ruff/pull/26886 fixes this, so I assume the odd error message is an artifact of the way the call binding currently fails. But we can leave it open, @dhruvmanila will know if this requires anything additional.

**Comment by dhruvmanila** [comment]
I can confirm this is part of the call-binding support for TVT because currently we infer `*args: *Ts` as `*args: Unknown` which is the return `Callable` type of the `subscribe` decorator.

**Comment by charliermarsh** [comment]
It looks like this was fixed by https://github.com/astral-sh/ruff/pull/26886.
